packages/headless/src/controllers/commerce/product-listing/facets/headless-product-listing-numeric-facet.ts

```diff
@@ -0,0 +1,37 @@
+import {CommerceEngine} from '../../../../app/commerce-engine/commerce-engine';
+import {fetchProductListing} from '../../../../features/commerce/product-listing/product-listing-actions';
+import {loadReducerError} from '../../../../utils/errors';
+import {CoreCommerceFacetOptions} from '../../facets/core/headless-core-commerce-facet';
+import {
+  CommerceNumericFacet,
+  buildCommerceNumericFacet,
+} from '../../facets/core/numeric/headless-commerce-numeric-facet';
+import {loadProductListingReducer} from '../utils/load-product-listing-reducers';
+
+export type ProductListingNumericFacetOptions = Omit<
+  CoreCommerceFacetOptions,
+  | 'fetchResultsActionCreator'
+  | 'toggleSelectActionCreator'
+  | 'toggleExcludeActionCreator'
+>;
+
+export type ProductListingNumericFacetBuilder =
+  typeof buildProductListingNumericFacet;
+
+export function buildProductListingNumericFacet(
+  engine: CommerceEngine,
+  options: ProductListingNumericFacetOptions
+): CommerceNumericFacet {
+  const coreController = buildCommerceNumericFacet(engine, {
+    ...options,
+    fetchResultsActionCreator: fetchProductListing,
+  });
+
+  if (!loadProductListingReducer(engine)) {
+    throw loadReducerError;
+  }
+
+  return {
+    ...coreController,
+  };
```

Comment:
Is there a reason we load the reducers after creating the controller? From what I can tell, we usually load the reducers first.

Doing so also allows us to return the controller right away. 👀 
```suggestion
  if (!loadProductListingReducer(engine)) {
    throw loadReducerError;
  }
  return buildCommerceNumericFacet(engine, {
    ...options,
    fetchResultsActionCreator: fetchProductListing,
  });
```
Same comment applies to `buildProductListingRegularFacet`

Reply:
No reason, no! It's an artefact of multiple rounds of refactoring 😁